storage/interceptors/broker_update_catalog_interceptor.go

```diff
@@ -78,6 +80,11 @@ func (c *brokerUpdateCatalogInterceptor) OnTxUpdate(f storage.InterceptUpdateOnT
 
 		oldBroker.Services = existingServiceOfferingsWithServicePlans.ServiceOfferings
 
+		err = reuseReferenceInstancePlans(oldObj.(*types.ServiceBroker), newObj.(*types.ServiceBroker))
```

Comment:
we should probably extract this code from broker_Catalog_create to a shared function and re-use in broker update logic as well right? 

```
if servicePlan.IsShareablePlan() {
				if !isBindablePlan(service, servicePlan) {
					return util.HandleInstanceSharingError(util.ErrPlanMustBeBindable, servicePlan.Name)
				}
				sharedPlanFound = true
			}
		}
		if sharedPlanFound {
			referencePlan := generateReferencePlanObject(service.ID)
			service.Plans = append(service.Plans, referencePlan)
			// Adds OSB spec properties only
			referencePlanOSBObj := convertReferencePlanObjectToOSBPlan(referencePlan)
			// The path should append reference plan into service plans json
			catalogJsonPath := fmt.Sprintf("services.%d.plans.-1", serviceIndex)
			catalogJson, err := sjson.SetBytes(broker.Catalog, catalogJsonPath, referencePlanOSBObj)
			if err != nil {
				return err
			}
			broker.Catalog = catalogJson
		}
```